### PR TITLE
AEA-1303 reconnect and retries for p2p_libp2p_client connection

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

reconnect and retries for p2p_libp2p_client connection

## Fixes

https://github.com/fetchai/agents-aea/issues/2154

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p_client/connection.py [hunk 1]
```diff
@@ -75,6 +74,8 @@ class P2PLibp2pClientConnection(Connection):
 
     connection_id = PUBLIC_ID
 
+    CONNECT_RETRIES = 3
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p_client/connection.py:77]: Please set this as default and also make configurable via yaml configs.

File: packages/fetchai/connections/p2p_libp2p_client/connection.py [hunk 2]
```diff
@@ -314,26 +365,34 @@ async def _send(self, data: bytes) -> None:
...
-            if not data:  # pragma: no cover
-                return None
-            return data
+            return await self._read_message_from_reader()
         except ConnectionError as e:  # pragma: nocover
-            self.logger.info(f"Connection error: {e}")
-            return None
+            self.logger.error(f"Connection error: {e}. Try to reconnect and read again")
+            await self._perform_connection_to_node()
+            return await self._read_message_from_reader()
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p_client/connection.py:390]: Will this work if the previous read experienced a disconnection?

**Review by DavidMinarsch — COMMENTED** [review]
Please also fix the broken tests.

File: packages/fetchai/connections/p2p_libp2p_client/connection.py (hunk 2 above)
Comment (solarw) [reply packages/fetchai/connections/p2p_libp2p_client/connection.py:390]: it sets a completely new connection, so will work

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**